src/test/kotlin/pl/komjago/ticketapp/services/BookingServiceTests.kt

```diff
@@ -0,0 +1,596 @@
+package pl.komjago.ticketapp.services
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import pl.komjago.ticketapp.controllers.booking.dto.BookedSeatInfo
+import pl.komjago.ticketapp.controllers.booking.dto.ChooseScreeningOutput
+import pl.komjago.ticketapp.controllers.booking.dto.GetScreeningsInput
+import pl.komjago.ticketapp.controllers.booking.dto.GetScreeningsOutput
+import pl.komjago.ticketapp.controllers.booking.dto.MakeReservationInput
+import pl.komjago.ticketapp.controllers.booking.dto.MakeReservationOutput
+import pl.komjago.ticketapp.controllers.booking.dto.MovieInfo
+import pl.komjago.ticketapp.controllers.booking.dto.RoomInfo
+import pl.komjago.ticketapp.controllers.booking.dto.ScreeningInfo
+import pl.komjago.ticketapp.controllers.booking.dto.SeatInfo
+import pl.komjago.ticketapp.controllers.booking.dto.TicketTypeInfo
+import pl.komjago.ticketapp.entity.Movie
+import pl.komjago.ticketapp.entity.Reservation
+import pl.komjago.ticketapp.entity.Room
+import pl.komjago.ticketapp.entity.Screening
+import pl.komjago.ticketapp.entity.Seat
+import pl.komjago.ticketapp.entity.Ticket
+import pl.komjago.ticketapp.entity.TicketType
+import pl.komjago.ticketapp.repository.ReservationRepository
+import pl.komjago.ticketapp.repository.ScreeningRepository
+import pl.komjago.ticketapp.repository.SeatRepository
+import pl.komjago.ticketapp.repository.TicketRepository
+import pl.komjago.ticketapp.repository.TicketTypeRepository
+import pl.komjago.ticketapp.util.toZloty
+import java.time.LocalDateTime
+import java.util.*
+import javax.persistence.EntityNotFoundException
+
+class BookingServiceTests {
+
+    private val reservationRepository: ReservationRepository = mockk()
+    private val ticketRepository: TicketRepository = mockk()
+    private val seatRepository: SeatRepository = mockk()
+    private val ticketTypeRepository: TicketTypeRepository = mockk()
+    private val screeningRepository: ScreeningRepository = mockk()
+
+    private val bookingService: BookingService =
+            BookingServiceImpl(reservationRepository,
+                    ticketRepository,
+                    seatRepository,
+                    ticketTypeRepository,
+                    screeningRepository,
+                    15
+            )
+
+    @Test
+    fun `get screenings with GetScreeningsInput returns GetScreeningsOutput sorted by title and screening time`() {
+        val input = GetScreeningsInput(LocalDateTime.MIN, LocalDateTime.MAX)
```

Comment:
czy nie warto by było się zabezpieczyć przed wyciąganiem takich zakresów dat? to wyciągnie całą db

Reply:
Racja, na przyszłość trzeba zabezpieczyć.